src/live-programming.rst

```diff
@@ -16,6 +16,379 @@ REPLs
+
+.. container:: bib-item
+
+  .. bibliography:: data-analysis-environments.bib
+    :filter: key == 'donoho:2017'
+
+  This article reflects on the past 50 years of history of data science. It introduces how the field of data science emerged as a superset of statistics and machine learning, driven by commercial rather than intellectual developments. It also visions how the field of data science will grow in the next 50 years.
+
+Data science has grown rapidly over the last decade as datasets have increased rapidly in size and ubiquity. 
+Kery and Myers :cite:`kery:2017:exploring` observe that data science practices are a form of exploratory programming, with the results of previous steps determining the next steps that the programmer will take.
+Muller et al. :cite:`muller:2019` investigated how data science workers engaged with data.
+They examined the workflow of data science by interviewing 21 data science professionals with diverse backgrounds and classifying how they work with data.
+They described the data science practices as data acquisition, cleaning and integrating and engineering features.
+
+.. container:: bib-item
+
+  .. bibliography:: data-analysis-environments.bib
+    :filter: key == 'kery:2017:exploring'
+
+  This paper reviews and describes what exploratory programming is. It outlines four dimensions to study exploratory programming tasks: applications, required code quality, ease of difficulty of exploration, and the exploration process.  
+
+
+.. container:: bib-item
+
+  .. bibliography:: data-analysis-environments.bib
+    :filter: key == 'muller:2019'
+
+  This paper describes the practice of data science with a focus on how data science workers engage with data. It reports five approaches to data: data as given; as captured; as curated; as designed; and as created.
+
+This chapter reflects on the challenges in the common practice of data analysis, and reports novel designs that address the challenges.
+The rest of the chapter is organized as the following:
+1) the first part of the article introduces novel tools and systems that assist with data collection, cleaning, and analyzing,
+2) the second part of the article introduces computational notebooks, a common tool for documenting and sharing the process of data science.
+
+Tools for Data Collection
+--------------------------
+Data can be collected from a variety of sources.
+For example, data can be downloaded as a public dataset from online repositories, or generated by physical sensors (weather data, personal health data), or collected from existing websites (flight prices, stock prices), or manually crafted (the grade of students' assignments).
+In data science education, there is an increasing call to encourage students to collect and explore authentic data that are meaningful to themselves rather than providing them a standard dataset.
+The common practice for data science workers to collect data from an existing website is using web scraping tools.
```

Comment:
These two sentences should introduce Web Scraping section, since that is not the only way to collect data.